Lib/test/test_wsgiref.py

```diff
@@ -525,6 +525,40 @@ def testExtras(self):
             '\r\n'
         )
 
+    validation_cases = (
```

Comment:
Please add a comment mentioning bpo-11671.

Rather than an hardcoded list, why not building such list using a few invalid characters. For example for header names:

```
invalid_names = []
invalid_name_chars = ['\r', '\n', '\r\n', ':']
for invalid_chars in invalid_name_chars:
  invalid_names.append(f'{invalid_chars}InvalidName')
  invalid_names.append(f'Invalid{invalid_chars}Name')
  invalid_names.append(f'InvalidName{invalid_chars}')
invalid_names.append(' InvalidName')
```

Are whitespaces (" ") and colon (":") only invalid at the beginning of the name?